packages/vaadin-grid/src/vaadin-grid-selection-column.js

```diff
@@ -197,15 +120,78 @@ class GridSelectionColumnElement extends GridColumnElement {
   connectedCallback() {
     super.connectedCallback();
     if (this._grid) {
-      this._grid.addEventListener('active-item-changed', this._boundOnActiveItemChanged);
-      this._grid.addEventListener('data-provider-changed', this._boundOnDataProviderChanged);
-      this._grid.addEventListener('filter-changed', this._boundOnSelectedItemsChanged);
-      this._grid.addEventListener('selected-items-changed', this._boundOnSelectedItemsChanged);
+      this._grid.addEventListener('active-item-changed', this.__boundOnActiveItemChanged);
+      this._grid.addEventListener('data-provider-changed', this.__boundOnDataProviderChanged);
+      this._grid.addEventListener('filter-changed', this.__boundOnSelectedItemsChanged);
+      this._grid.addEventListener('selected-items-changed', this.__boundOnSelectedItemsChanged);
     }
   }
 
+  /**
+   * Renders the Select All checkbox to the header cell
+   *
+   * @private
+   */
+  __defaultHeaderRenderer(root, _column) {
+    let checkbox = root.firstElementChild;
+    if (!checkbox) {
+      checkbox = document.createElement('vaadin-checkbox');
+      checkbox.setAttribute('aria-label', 'Select All');
+      checkbox.classList.add('vaadin-grid-select-all-checkbox');
+      checkbox.addEventListener('checked-changed', this.__onSelectAllCheckedChanged.bind(this));
+      root.appendChild(checkbox);
+    }
+
+    const checked = this.__isChecked(this.selectAll, this.__indeterminate);
+    checkbox.__rendererChecked = checked;
+    checkbox.checked = checked;
+    checkbox.indeterminate = this.__indeterminate;
+
+    if (this.__selectAllHidden) {
+      checkbox.setAttribute('hidden', 'hidden');
+    } else {
+      checkbox.removeAttribute('hidden');
+    }
```

Comment:
Would `checkbox.hidden = this.__selectAllHidden` work here?

Reply:
Good point, thanks!